debug_toolbar/store.py

```diff
@@ -140,5 +141,91 @@ def panels(cls, request_id: str) -> Any:
             yield panel, deserialize(data)
 
 
+class DatabaseStore(BaseStore):
+    @classmethod
+    def _cleanup_old_entries(cls):
+        """
+        Enforce the cache size limit - keeping only the most recently used entries
+        up to RESULTS_CACHE_SIZE.
+        """
+        # Get the cache size limit from settings
+        cache_size = dt_settings.get_config()["RESULTS_CACHE_SIZE"]
+
+        # Determine which entries to keep (the most recent ones up to cache_size)
+        keep_ids = list(
+            HistoryEntry.objects.all()[:cache_size].values_list("request_id", flat=True)
+        )
```

Comment:
```suggestion
        # Determine which entries to keep
        keep_ids = cls.request_ids()
```

There's some code we can re-use. This should reduce what the test needs to do.